clients/input_copy.go

```diff
@@ -95,6 +92,89 @@ func (s *InputCopy) CopyInputToS3(requestID string, inputFile *url.URL) (inputVi
 	return
 }
 
+func isHLSInput(inputFile *url.URL) bool {
+	ext := strings.LastIndex(inputFile.Path, ".")
+	if ext == -1 {
+		return false
+	}
+	return inputFile.Path[ext:] == ".m3u8"
+}
+
+func getDestLoc(srcManifestUrl, dstTransferUrl *url.URL, srcSegmentUrl string) string {
```

Comment:
@leszko, lmk if there's a cleaner way for this implementation - it works but I think it could be simpler and I got lost in the world of golang's url lib and resorted to a very basic implementation. 

As an example, this is what this function is supposed to do given the following args: 

srcManifestUrl := `https://link.storjshare.io/raw/jwbmejfygkb6mdrarfplvc46zeka/cakesource/rec/b40efldl0rzdcm7b/2023.04.19.14.04.01/output.m3u8`

srcSegmentUrl := `https://link.storjshare.io/raw/jwbmejfygkb6mdrarfplvc46zeka/cakesource/rec/b40efldl0rzdcm7b/2023.04.19.14.04.01/blah/hello/world/3.ts`

dstTransferUrl := `s3+https://USER:PASS@storage.googleapis.com/lp-us-catalyst-vod-monster/hls/source/hcfecbea/transfer/output.m3u8`

The function should return the target transfer directory where *each* segment from the storj src URL can be copied to. In this case, the function should return: 
```s3+https://GOOG1ECL5LAFILBTTALJ6EFZTFJQSBC6QVRJWXEKROJW6Y2R7RZ25WPE2VNVA:foWv%2F3xcMT1+q4g1SLDCYtJ47y3USZ%2FMKGD+HmbK@storage.googleapis.com/lp-us-catalyst-vod-monster/hls/source/hcfecbea/transfer/blah/hello/world/3.ts```

Reply:
Done